### ISSUE TITLE
`nodejs` requirement?

### ISSUE THREAD

**Issue by kcpevey** [pyviz/EarthSim#140]
I pulled the latest master from earthsim and holoviews this morning to test out linking tables.   
holoviews                 1.10.6.post3+g435f00cc  
  
I ran the `Linking_Plots.ipynb` in the holoviews repo without any errors on `hv.extension('bokeh')`. However, when I ran the `Annotators.ipynb`, it failed on `hv.extension('bokeh')` because I didn't have `nodejs`. This is the same environment that I've been using so I'm not sure if this is a new requirement?   
  
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-1-626bc622167c> in <module>()
      7 from earthsim.annotators import (PolyAnnotator, PointAnnotator, PolyAndPointAnnotator,
      8                                  GeoAnnotator,  PointWidgetAnnotator)
----> 9 hv.extension('bokeh')

C:\ProgramData\Anaconda3\envs\ers\lib\site-packages\param\parameterized.py in __new__(class_, *args, **params)
   1889         inst = class_.instance()
   1890         inst._set_name(class_.__name__)
-> 1891         return inst.__call__(*args,**params)
   1892 
   1893     def __call__(self,*args,**kw):

c:\projects\ers\github\holoviews\master\holoviews\holoviews\ipython\__init__.py in __call__(self, *args, **params)
    167         Renderer.load_nb()
    168         for r in [r for r in resources if r != 'holoviews']:
--> 169             Store.renderers[r].load_nb(inline=p.inline)
    170 
    171         if hasattr(ip, 'kernel') and not loaded:

c:\projects\ers\github\holoviews\master\holoviews\holoviews\plotting\bokeh\renderer.py in load_nb(cls, inline)
    386         LOAD_MIME_TYPE = bokeh.io.notebook.LOAD_MIME_TYPE
    387         bokeh.io.notebook.LOAD_MIME_TYPE = MIME_TYPES['jlab-hv-load']
--> 388         load_notebook(hide_banner=True, resources=INLINE if inline else CDN)
    389         bokeh.io.notebook.LOAD_MIME_TYPE = LOAD_MIME_TYPE
    390         bokeh.io.notebook.curstate().output_notebook()

C:\ProgramData\Anaconda3\envs\ers\lib\site-packages\bokeh\io\notebook.py in load_notebook(resources, verbose, hide_banner, load_timeout)
    397     _NOTEBOOK_LOADED = resources
    398 
--> 399     custom_models_js = bundle_all_models()
    400 
    401     nb_js = _loading_js(resources, element_id, custom_models_js, load_timeout, register_mime=True)

C:\ProgramData\Anaconda3\envs\ers\lib\site-packages\bokeh\util\compiler.py in bundle_all_models()
    512     bundle = _bundle_cache.get(key, None)
    513     if bundle is None:
--> 514         _bundle_cache[key] = bundle = bundle_models(Model.model_class_reverse_map.values()) or ""
    515     return bundle

C:\ProgramData\Anaconda3\envs\ers\lib\site-packages\bokeh\util\compiler.py in bundle_models(models)
    398     for model in ordered_models:
    399         impl = model.implementation
--> 400         compiled = nodejs_compile(impl.code, lang=impl.lang, file=impl.file)
    401 
    402         if "error" in compiled:

C:\ProgramData\Anaconda3\envs\ers\lib\site-packages\bokeh\util\compiler.py in nodejs_compile(code, lang, file)
    187 def nodejs_compile(code, lang="javascript", file=None):
    188     compilejs_script = join(bokehjs_dir, "js", "compiler.js")
--> 189     output = _run_nodejs([compilejs_script], dict(code=code, lang=lang, file=file))
    190     return AttrDict(json.loads(output))
    191 

C:\ProgramData\Anaconda3\envs\ers\lib\site-packages\bokeh\util\compiler.py in _run_nodejs(argv, input)
    166 
    167 def _run_nodejs(argv, input=None):
--> 168     return _run(_nodejs_path(), argv, input)
    169 
    170 def _run_npmjs(argv, input=None):

C:\ProgramData\Anaconda3\envs\ers\lib\site-packages\bokeh\util\compiler.py in _nodejs_path()
    147     global _nodejs
    148     if _nodejs is None:
--> 149         _nodejs = _detect_nodejs()
    150     return _nodejs
    151 

C:\ProgramData\Anaconda3\envs\ers\lib\site-packages\bokeh\util\compiler.py in _detect_nodejs()
    139     version = ".".join(map(str, nodejs_min_version))
    140     raise RuntimeError('node.js v%s or higher is needed to allow compilation of custom models ' % version +
--> 141                        '("conda install nodejs" or follow https://nodejs.org/en/download/)')
    142 
    143 _nodejs = None

RuntimeError: node.js v6.10.0 or higher is needed to allow compilation of custom models ("conda install nodejs" or follow https://nodejs.org/en/download/)
```

**Comment by philippjfr** [comment]
>This is the same environment that I've been using so I'm not sure if this is a new requirement?

Quite possible, the new clear and snapshot tools have to be compiled using nodejs. Will have to add it to the environment.

**Comment by ceball** [comment]
Yes, I noticed this, and it's in my PR to fix up the environment. So will soon be in master...

**Comment by ceball** [comment]
(I was talking about earthsim's environment.)

**Comment by ceball** [comment]
Should have been fixed by #139
